multimodal/src/autogluon/multimodal/data/preprocess_dataframe.py

```diff
@@ -306,6 +306,10 @@ def _fit_x(self, X: pd.DataFrame):
         if self._fit_x_called:
             raise RuntimeError("fit_x() has been called. Please create a new preprocessor and call it again!")
         self._fit_x_called = True
+        # Creating deep copy of the DataFrame, which allows writable buffer to be created for the new df
+        # This is needed for 1.4.1 < scikit-learn < 1.5.0, the versions 1.4.0 and 1.5.1 do not need a writable buffer
+        X = X.copy(deep=True)
```

Comment:
My question is if we should do a deep copy (presumably slow?) just for <1.5.0.

Reply:
should not be the case